### PR TITLE
[MIG] contract_price_revision: Migration to 12.0

### PR DESCRIPTION
**Pull request by carlosdauden**
@Tecnativa TT19362

### CODE REVIEW

**Review by sbejaoui — APPROVED** [review]
LGTM thank you!

File: contract_price_revision/models/contract_line.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        digits=dp.get_precision('Product Price'),
+        string='Variation %',
+    )
+
+    @api.depends('price_unit', 'predecessor_contract_line_id.price_unit')
+    def _compute_variation_percent(self):
+        for line in self:
+            if line.price_unit and line.previous_price:
+                line.variation_percent = (
+                    (line.price_unit / line.previous_price - 1) * 100)
```
Comment (Tardo) [inline contract_price_revision/models/contract_line.py]: This can throw an ZeroDivisionError error... no?

**Review by Tardo — APPROVED** [review]
:+1: LGTM. Code review

File: contract_price_revision/models/account_analytic_invoice_line.py [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-class AccountAnalyticINvoiceLine(models.Model):
-    _inherit = "account.analytic.invoice.line"
-
-    date_start = fields.Date(
-        string='Start Date',
-    )
-    date_end = fields.Date(
-        string='End Date',
-    )
-    previous_revision_id = fields.Many2one(
```
Comment (pedrobaeza) [inline contract_price_revision/models/account_analytic_invoice_line.py]: We should convert this field in new `predecessor_contract_line_id` field in a migration script. And what about the dates?

File: contract_price_revision/tests/test_contract_price_revision.py [hunk 3]
```diff
@@ -1,71 +1,43 @@
...
+        wizard = self.env['contract.price.revision.wizard'].create({
+            'date_start': '2018-02-15',
             'variation_percent': 100.0,
         })
         wizard.with_context(
             {'active_ids': [self.contract.id]}).action_apply()
 
     def test_contract_price_revision_wizard(self):
-        self.assertEqual(len(self.contract.recurring_invoice_line_ids.ids), 2)
+        self.acct_line.copy({'automatic_price': True})
```
Comment (pedrobaeza) [inline contract_price_revision/tests/test_contract_price_revision.py:20]: You can comment that this is for checking if this line is not versioned

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by astirpe** [comment]
@carlosdauden could you cherry-pick https://github.com/OCA/contract/pull/469/commits/db5c3d647225dde2b20214932a060aa279554b07 from https://github.com/OCA/contract/pull/469 ?

**Comment by pedrobaeza** [comment]
@carlosdauden please finish this.

**Review by rafaelbn — APPROVED** [review]
:+1: 

@carlosdauden please check last comments

**Comment by carlosdauden** [comment]
Changes done

File: contract_price_revision/models/account_analytic_invoice_line.py (hunk 2 above)
Comment (carlosdauden) [reply contract_price_revision/models/account_analytic_invoice_line.py]: Script created.
Date fields has the same name and the table was renamed keeping all fields and values.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**